test/test_tags.rb

```diff
@@ -13,7 +13,7 @@ def create_post(content, override = {}, converter_class = Jekyll::Converters::Ma
     site = Site.new(Jekyll.configuration)
 
     if override['read_posts']
-      site.read_posts('')
+      site.posts += PostReader.new(site).read('')
```

Comment:
I replaced all the `+=` occurrences with concatenations. Apparently `+=` creates a new array every time it is used, while `concat` simply appends the new elements to the old array.

[Reference](http://stackoverflow.com/questions/23763215/ruby-array-concat-versus-speed)